shared/actions/fs/index.js

```diff
@@ -462,23 +462,24 @@ function* ignoreFavoriteSaga(action: FsGen.FavoriteIgnorePayload): Saga.SagaGene
 //   Text/HTML;Charset="utf-8"
 //   text/html; charset="utf-8"
 // The last part is optional, so if `;` is missing, it'd be just the mimetype.
-const extractMimeTypeFromContentType = (contentType: string): string => {
+const extractMimeTypeFromContentType = (contentType, disposition: string): Types.Mime => {
   const ind = contentType.indexOf(';')
-  return (ind > -1 ? contentType.slice(0, ind) : contentType).toLowerCase()
+  const mimeType = (ind > -1 ? contentType.slice(0, ind) : contentType).toLowerCase()
+  return Constants.makeMime({mimeType, displayPreview: disposition !== 'attachment'})
```

Comment:
I don't think this would cover this form:

```
Content-Disposition: attachment; filename="filename.jpg"
```

Also, it seems better to filter the other way around, i.e. `disposition === 'inline`, so that we don't accidentally display non-inline and non-attachment stuff inline. WDYT?

Reply:
Old server provides no disposition, so I'd like to default to more previews if there is a weird mismatch.

Reply:
Will handle more complex dispositions correctly.

Reply:
> Old server provides no disposition, so I'd like to default to more previews if there is a weird mismatch.

If we merge both together, this should be super rare. Also, we could probably temporarily default to `inline` if it's empty until more releases have gone out. According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition it can also be `form-data`. On the other hand we don't use any of those on the Go side so it's probably fine. It'd be nice to have a comment either here or on the Go side explaining it though.